packages/tx/src/legacyTransaction.ts

```diff
@@ -50,73 +44,65 @@ export default class Transaction {
     return this.fromValuesArray(values, opts)
   }
 
-  public static fromValuesArray(values: Buffer[], opts?: TxOptions) {
+  public static fromValuesArray(values: Buffer[], opts: TxOptions = {}) {
     if (values.length !== 6 && values.length !== 9) {
       throw new Error(
         'Invalid transaction. Only expecting 6 values (for unsigned tx) or 9 values (for signed tx).'
       )
     }
 
-    const [nonce, gasPrice, gasLimit, to, value, data, v, r, s] = values
-
-    const emptyBuffer = Buffer.from([])
-
-    return new Transaction(
-      {
-        nonce: new BN(nonce),
-        gasPrice: new BN(gasPrice),
-        gasLimit: new BN(gasLimit),
-        to: to && to.length > 0 ? new Address(to) : undefined,
-        value: new BN(value),
-        data: data ?? emptyBuffer,
-        v: v && !v.equals(emptyBuffer) ? new BN(v) : undefined,
-        r: r && !r.equals(emptyBuffer) ? new BN(r) : undefined,
-        s: s && !s.equals(emptyBuffer) ? new BN(s) : undefined,
-      },
-      opts
-    )
+    // If length is not 6, it has length 9. If v/r/s are empty Buffers, it is still an unsigned transaction
+    // This happens if you get the RLP data from `raw()`
+    if (values.length === 6 || values.length == 9) {
+      const [nonce, gasPrice, gasLimit, to, value, data, v, r, s] = values
+
+      const emptyBuffer = Buffer.from([])
+
+      return new LegacyTransaction(
+        {
+          nonce: new BN(nonce),
+          gasPrice: new BN(gasPrice),
+          gasLimit: new BN(gasLimit),
+          to: to && to.length > 0 ? new Address(to) : undefined,
+          value: new BN(value),
+          data: data ?? emptyBuffer,
+          v: v !== undefined && !v.equals(emptyBuffer) ? new BN(v) : undefined,
+          r: r !== undefined && !r.equals(emptyBuffer) ? new BN(r) : undefined,
+          s: s !== undefined && !s.equals(emptyBuffer) ? new BN(s) : undefined,
```

Comment:
TODO: check if this is compliant with the latest check to these v/r/s values, this might have gone wrong on the rebase.